networkx/generators/tests/test_line.py

```diff
@@ -275,3 +275,86 @@ def test_line_different_node_types(self):
         H = nx.line_graph(G)
         J = nx.inverse_line_graph(H)
         assert nx.is_isomorphic(G, J)
+
+    def test_line_vertex_partition_appearances(self):
+        G = nx.Graph()
+        G_vertices = [
+            (1, 2),
+            (2, 3),
+            (3, 4),
+            (1, 4),
+            (4, 5),
+            (5, 1),
+            (1, 6),
+            (6, 7),
+            (7, 1),
+        ]
+        G.add_edges_from(G_vertices)
+        pytest.raises(nx.NetworkXError, nx.inverse_line_graph, G)
+
+    def test_line_inverse_edgeless_graph(self):
+        G = nx.Graph()
+        G_nodes = [1, 2, 3, "tree", "branches"]
+        G.add_nodes_from(G_nodes)
+        pytest.raises(nx.NetworkXError, nx.inverse_line_graph, G)
+
+    def test_line_graph_exists(self):
+        G = nx.Graph()
+        G_vertices = [(1, 2), (2, 3), (3, 1), (1, 4), (4, 5), (5, 1), (1, 6)]
+        G.add_edges_from(G_vertices)
+        pytest.raises(nx.NetworkXError, nx.inverse_line_graph, G)
+
+    def test_line_triangle_vertex_not_in_graph(self):
+        G = nx.Graph()
+        pytest.raises(nx.NetworkXError, line._triangles, G, (1, 2))
+
+    def test_line_edge_not_exists(self):
+        G = nx.barabasi_albert_graph(10, 5)
+        pytest.raises(nx.NetworkXError, line._triangles, G, (5, 11))
+
+    def test_line_odd_triangle_vertex_not_in_graph(self):
+        G = nx.Graph()
+        G_edges = [(1, 2), (2, 3), (3, 1)]
+        G.add_edges_from(G_edges)
+        pytest.raises(nx.NetworkXError, line._odd_triangle, G, [1, 2, 4])
+
+    def test_line_odd_triangle_edges_not_in_graph(self):
+        G = nx.Graph()
+        G_edges = [(1, 2), (2, 3), (3, 2)]
+        G.add_edges_from(G_edges)
+        pytest.raises(nx.NetworkXError, line._odd_triangle, G, [1, 2, 3])
+
+    def test_line_starting_edge_not_in_graph(self):
+        G = nx.Graph()
+        G_edges = [(1, 2), (2, 3), (3, 1)]
+        G.add_edges_from(G_edges)
+        pytest.raises(nx.NetworkXError, line._select_starting_cell, G, (1, 4))
+
+    def test_line_rs(self):
+        G = nx.Graph()
```

Comment:
Thanks for all the tests @nsengiyumva-wilberforce!

Could you also add some sort of descriptions what exactly is the function supposed to test?
This description could be comment in the function. Something like:

``` python
def test_line_rs(self):
    # Test .................
    ...
    ...
```

Reply:
I have commented all the methods that do not have direct names to what they are doing!


thanks big

Reply:
I am going to adhere to this practice of commenting functions that do not have clear names


Glad I have learnt this, thank you very much @MridulS